ansible/ansible
# New module: Support manipulating XML files [files/xml]

##### SUMMARY
This is another attempt to get the xml module upstream.
https://github.com/cmprescott/ansible-xml/

Prior submissions at: #16147, ansible/ansible-modules-core#4112 and ansible/ansible-modules-core#4122

This PR includes:
- The code has been re-licensed to GPLv3+. cmprescott/ansible-xml#113
- 55 integration tests (tested in run-mode and check-mode)
- Add return values to documentation
- Added python 3 support (revised unicode support)
- Improved and simplified check-mode support
- Fixed one idempotency bug
- Fixed a bug related to check-mode support (incorrectl reported no change)

Once merged we can collaborate on the module and tests in this project:
- Migrate and fix [open issues](https://github.com/cmprescott/ansible-xml/issues/)
- Merge [existing PRs](https://github.com/cmprescott/ansible-xml/pulls/)
- Add diff support !!  --  *Ideal in combination with check-mode*
- Add backup support

This fixes cmprescott/ansible-xml#52,
and this fixes cmprescott/ansible-xml#74,
and this fixes cmprescott/ansible-xml#75,
but his also fixes cmprescott/ansible-xml#108,
and this fixes cmprescott/ansible-xml#109,
and this fixes cmprescott/ansible-xml#116,

Due to the addition of a namespace example and a documentation note (as stated in cmprescott/ansible-xml#118),
this fixes cmprescott/ansible-xml#102,
and it fixes cmprescott/ansible-xml#107,
and fixes cmprescott/ansible-xml#112,
and also fixes cmprescott/ansible-xml#117,
and fixes cmprescott/ansible-xml#118.

Due to an improved lxml version check,
this fixes cmprescott/ansible-xml#114
and fixes cmprescott/ansible-xml#115.,

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

xml

##### ANSIBLE VERSION
v2.4
